javaslang/src/test/java/javaslang/collection/TreeMapTest.java

```diff
@@ -122,6 +126,22 @@ public void shouldWrapMap() {
         assertThat(TreeMap.ofAll(source)).isEqualTo(emptyIntInt().put(1, 2).put(3, 4));
     }
 
+    // -- ofAll
+
+    @Test
+    public void shouldCreateKeyComparatorForJavaUtilMap() {
+        final TreeMap<String, Integer> actual = TreeMap.ofAll(mapOfTuples(Tuple.of("c", 0), Tuple.of("a", 0), Tuple.of("b", 0)).toJavaMap());
+        final List<String> expected = List.of("a", "b", "c");
+        assertThat(actual.keySet().toList()).isEqualTo(expected);
```

Comment:
@paplorinc This works on my current RedBlackTree branch:

```java
    @Test
    public void shouldCreateKeyComparatorForJavaUtilMap() {
        final TreeMap<String, Integer> actual = TreeMap.ofAll(List.of(Tuple.of("c", 0), Tuple.of("a", 0), Tuple.of("b", 0)).toJavaMap(t -> t));
        final List<Tuple2<String, Integer>> expected = List.of(Tuple.of("a", 0), Tuple.of("b", 0), Tuple.of("c", 0));
        assertThat(actual.toList()).isEqualTo(expected);
    }
```

Could you replace your test with it, please. No bug here as far as I can see...